Build failed on Windows11 with llvm-mingw

### Tested versions

4.6-beta d743736f84fb668bcdfb5bf86258e483f35c6fec

### System information

Windows 11 25H2

### Issue description

Building with:
- llvm-mingw251202 also with a previous version
- scons 4.10/4.9.1
gives me a compile errror:
```
scons use_llvm=yes use_mingw=yes compiledb=yes
scons: Reading SConscript files ...
Automatically detected platform: windows
Auto-detected 20 CPU cores available for build parallelism. Using 19 cores by default. You can override it with the `-j` or `num_jobs` arguments.
Building for platform "windows", architecture "x86_64", target "editor".
scons: done reading SConscript files.
scons: Building targets ...
[ 24%] Compiling modules\navigation_2d\2d\nav_mesh_generator_2d.cpp ...
[ 29%] ERROR: In file included from modules\navigation_2d\2d\nav_mesh_generator_2d.cpp:39:
In file included from ./thirdparty/clipper2/include/clipper2/clipper.h:13:
thirdparty\clipper2\include/clipper2/clipper.core.h:598:53: error: no member named 'back_inserter' in namespace 'std'
  598 |     std::transform(path.cbegin(), path.cend(), std::back_inserter(result),
      |                                                     ^~~~~~~~~~~~~
thirdparty\clipper2\include/clipper2/clipper.core.h:607:55: error: no member named 'back_inserter' in namespace 'std'
  607 |     std::transform(paths.cbegin(), paths.cend(), std::back_inserter(result),
      |                                                       ^~~~~~~~~~~~~
2 errors generated.

[ 45%] Compiling editor\scene\2d\sprite_2d_editor_plugin.cpp ...
scons: *** [bin\obj\modules\navigation_2d\2d\nav_mesh_generator_2d.windows.editor.x86_64.llvm.o] Error 1
ERROR: In file included from editor\scene\2d\sprite_2d_editor_plugin.cpp:50:
In file included from ./thirdparty/clipper2/include/clipper2/clipper.h:13:
thirdparty\clipper2\include/clipper2/clipper.core.h:598:53: error: no member named 'back_inserter' in namespace 'std'
  598 |     std::transform(path.cbegin(), path.cend(), std::back_inserter(result),
      |                                                     ^~~~~~~~~~~~~
thirdparty\clipper2\include/clipper2/clipper.core.h:607:55: error: no member named 'back_inserter' in namespace 'std'
  607 |     std::transform(paths.cbegin(), paths.cend(), std::back_inserter(result),
      |                                                       ^~~~~~~~~~~~~
2 errors generated.

scons: *** [bin\obj\editor\scene\2d\sprite_2d_editor_plugin.windows.editor.x86_64.llvm.o] Error 1
scons: building terminated because of errors.
```
After adding `#include <iterator>` to clipper.core.h, I managed to build it successfully.

### Steps to reproduce

Build the engine on Windows 11 with llvm-mingw

### Minimal reproduction project (MRP)

No mrp, just build the engine.

--- a/modules/text_server_adv/text_server_adv.h
+++ b/modules/text_server_adv/text_server_adv.h
@@ -95,7 +95,7 @@ using namespace godot;
 // Thirdparty headers.
 
 GODOT_GCC_WARNING_PUSH_AND_IGNORE("-Wshadow")
-#ifdef __EMSCRIPTEN__
+#if defined(__EMSCRIPTEN__) || __clang_major__ >= 21
 GODOT_CLANG_WARNING_PUSH_AND_IGNORE("-Wunnecessary-virtual-specifier")
 #endif
 
@@ -113,7 +113,7 @@ GODOT_CLANG_WARNING_PUSH_AND_IGNORE("-Wunnecessary-virtual-specifier")
 #include <unicode/utypes.h>
 
 GODOT_GCC_WARNING_POP
-#ifdef __EMSCRIPTEN__
+#if defined(__EMSCRIPTEN__) || __clang_major__ >= 21
 GODOT_CLANG_WARNING_POP
 #endif
 

--- a/thirdparty/README.md
+++ b/thirdparty/README.md
@@ -132,6 +132,7 @@ Files extracted from upstream source:
 Patches:
 
 - `0001-disable-exceptions.patch` (GH-80796)
+- `0002-llvm-21-header.patch` (GH-113850)
 
 
 ## cvtt

--- a/thirdparty/clipper2/include/clipper2/clipper.core.h
+++ b/thirdparty/clipper2/include/clipper2/clipper.core.h
@@ -13,6 +13,7 @@
 #include "clipper2/clipper.version.h"
 #include <cstdint>
 #include <vector>
+#include <iterator>
 #include <string>
 #include <iostream>
 #include <algorithm>

--- a/thirdparty/clipper2/patches/0002-llvm-21-header.patch
+++ b/thirdparty/clipper2/patches/0002-llvm-21-header.patch
@@ -1,0 +1,12 @@
+diff --git a/thirdparty/clipper2/include/clipper2/clipper.core.h b/thirdparty/clipper2/include/clipper2/clipper.core.h
+index 5a698e96f4..a68e87225e 100644
+--- a/thirdparty/clipper2/include/clipper2/clipper.core.h
++++ b/thirdparty/clipper2/include/clipper2/clipper.core.h
+@@ -13,6 +13,7 @@
+ #include "clipper2/clipper.version.h"
+ #include <cstdint>
+ #include <vector>
++#include <iterator>
+ #include <string>
+ #include <iostream>
+ #include <algorithm>

